src/components/data-exploration/heatmap/HeatmapMetadataTrackSettings.jsx

```diff
@@ -63,7 +48,7 @@ const HeatmapMetadataTrackSettings = (props) => {
     if (_.isEqual(trackData, newTrackData)) return;
 
     setTrackData(getUpdatedTrackData());
-  }, [cellSets.hierarchy]);
+  }, [hierarchy]);
```

Comment:
Can we depend on `selectedHierarchy` and not use `hierarchy` at all? I am not completely sure though, so let me know if you try it out and it doesn't work.